src/fsharp/DependencyManager.Integration.fs

```diff
@@ -0,0 +1,230 @@
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+type ImplDependencyManager() =
+    interface IDependencyManagerProvider with
+        member __.Name = "Impl library loader"
+        member __.ToolName = ""
+        member __.Key = "impl"
+        member __.ResolveDependencies(_targetFramework:string, _scriptDir: string, _scriptName: string, _packageManagerTextLines: string seq) = 
+            None,[]
+
+    interface System.IDisposable with
+        member __.Dispose() = ()
+
+let registeredDependencyManagers = lazy (
+    let defaultProviders =
+        [new ProjectDependencyManager() :> IDependencyManagerProvider
+         new RefDependencyManager() :> IDependencyManagerProvider
+         new ImplDependencyManager() :> IDependencyManagerProvider]
+    
+    let loadedProviders =
+        enumerateDependencyManagerAssembliesFromCurrentAssemblyLocation()
+        |> Seq.collect (fun a -> a.GetTypes())
+        |> Seq.choose ReflectionDependencyManagerProvider.InstanceMaker
+        |> Seq.map (fun maker -> maker ())
+    
+    defaultProviders
+    |> Seq.append loadedProviders
+    |> Seq.map (fun pm -> pm.Key, pm)
+    |> Map.ofSeq
+)
+
+let RegisteredDependencyManagers() = registeredDependencyManagers.Force()
+
+let createPackageManagerUnknownError packageManagerKey m =
+    let registeredKeys = String.Join(", ", RegisteredDependencyManagers() |> Seq.map (fun kv -> kv.Value.Key))
+    let searchPaths = assemblySearchPaths.Force()
+    Error(FSComp.SR.packageManagerUnknown(packageManagerKey, String.Join(", ", searchPaths), registeredKeys),m)
+
+let tryFindDependencyManagerInPath m (path:string) : ReferenceType =
+    try
+        match path.IndexOf(":") with
+        | -1 | 1 ->
+            ReferenceType.Library path
+        | _ ->
+            let managers = RegisteredDependencyManagers()
+            match managers |> Seq.tryFind (fun kv -> path.StartsWith(kv.Value.Key + ":" )) with
+            | None ->
+                errorR(createPackageManagerUnknownError (path.Split(':').[0]) m)
+                ReferenceType.UnknownType
+            | Some kv -> ReferenceType.RegisteredDependencyManager kv.Value
+    with
+    | e ->
+        errorR(Error(FSComp.SR.packageManagerError(e.Message),m))
+        ReferenceType.UnknownType
+
+let removeDependencyManagerKey (packageManagerKey:string) (path:string) = path.Substring(packageManagerKey.Length + 1).Trim()
+
+let tryFindDependencyManagerByKey m (key:string) : IDependencyManagerProvider option =
+    try
+        RegisteredDependencyManagers() |> Map.tryFind key
+    with 
+    | e -> 
+        errorR(Error(FSComp.SR.packageManagerError(e.Message),m))
+        None
+
+let resolve (packageManager:IDependencyManagerProvider) implicitIncludeDir fileName m packageManagerTextLines =
```

Comment:
Could we have /// comments on all new top-level ``let`` , ``type`` and ``module`` please?  thanks